File: semester 4/MPP/JFTS/FestivalDomain/src/main/java/ro/mpp/domain/ShowArtist.java
package ro.mpp.domain;

import jakarta.persistence.*;
import lombok.*;

@Data
@Entity
@NoArgsConstructor
@Table(name = "show_artist")
public class ShowArtist {
    @EmbeddedId
    private ShowArtistId id;

    @ManyToOne(fetch = FetchType.EAGER)
    @MapsId("showId")
    @JoinColumn(name = "show_id")
    private Show show;

    @ManyToOne(fetch = FetchType.EAGER)
    @MapsId("artistId")
    @JoinColumn(name = "artist_id")
    private Artist artist;

    public ShowArtist(Show show, Artist artist) {
        this.show = show;
        this.artist = artist;
        this.id = new ShowArtistId(show.getId(), artist.getId());
    }
}

File: semester 4/MPP/C#FTS/Model/Domain/ShowArtist.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace C_FTS.Domain
{
    public class ShowArtist : Entity<int>
    {
        public Show Show { get; private set; }
        public Artist Artist { get; private set; }

        public ShowArtist(Show show, Artist artist) : base(-1)
        {
            this.Show = show;
            this.Artist = artist;
        }

        public override string ToString()
        {
            return $"ShowArtist(Show = {Show}, Artist = {Artist})";
        }
    }
}
